### PR TITLE
[Bug 1152406] Make AAQ support iOS useragent.

### PR DESCRIPTION
**Pull request by mythmon**
This also include a couple lint and cleanup commits. I can break those out easily if needed.

r?

### CODE REVIEW

File: kitsune/sumo/static/sumo/js/aaq.js [hunk 1]
```diff
@@ -14,16 +14,17 @@
...
+      // Autofill the user agent
       $form.find('input[name="useragent"]').val(navigator.userAgent);
 
       // Only guess at OS, FF version, plugins if we are on the desktop
       // asking a firefox desktop question (or s/desktop/mobile/).
       if ((BrowserDetect.browser === 'fx' && self.isDesktopFF()) ||
       (BrowserDetect.browser === 'm' && self.isMobileFF()) ||
-      (BrowserDetect.browser === 'fxos' && self.isFirefoxOS())) {
+      (BrowserDetect.browser === 'fxos' && self.isFirefoxOS()) ||
+      (BrowserDetect.browser === 'fxios' && self.isFirefoxIOS())) {
```
Comment (willkg) [inline kitsune/sumo/static/sumo/js/aaq.js]: "FirefoxOS" and "FirefoxIOS" are so similar, I wonder if it behooves us to make the second one more different to make mistakes more obvious.

File: kitsune/sumo/static/sumo/js/aaq.js [hunk 2]
```diff
@@ -64,39 +65,40 @@
...
+          ['Mac OS X 10.4', /(Mac OS X 10.4)/i],
+          ['Mac OS X 10.5', /(Mac OS X 10.5)/i],
+          ['Mac OS X 10.6', /(Mac OS X 10.6)/i],
+          ['Mac OS X 10.7', /(Mac OS X 10.7)/i],
+          ['Mac OS X 10.8', /(Mac OS X 10.8)/i],
+          ['Mac OS X 10.9', /(Mac OS X 10.9)/i],
+          ['Mac OS X 10.10', /(Mac OS X 10.10)/i],
+          ['Mac OS', /(Mac_PowerPC)|(Macintosh)/i],
+          ['QNX', /QNX/i],
+          ['BeOS', /BeOS/i],
```
Comment (willkg) [inline kitsune/sumo/static/sumo/js/aaq.js]: Thank goodness we recognize BeOS.

File: kitsune/sumo/static/sumo/js/browserdetect.js [hunk 3]
```diff
@@ -130,6 +136,11 @@ var BrowserDetect = {
       identity: 'win'
     },
     {
+      string: navigator.userAgent,
+      subStrings: [/iPad|iPhone/],
```
Comment (willkg) [inline kitsune/sumo/static/sumo/js/browserdetect.js]: Does this catch iPod touch, too?

**Comment by willkg** [comment]
I skimmed this and it looks ok. Figured I'd skim it since I just added i(Pad|Pod touch|Phone) support for Fjord.

I don't know how to test this out, though, so someone else should look through it, too.

File: kitsune/sumo/static/sumo/js/aaq.js (hunk 1 above)
Comment (rehandalal) [inline kitsune/sumo/static/sumo/js/aaq.js]: I was thinking the same thing. It took me a second to realize they were different. Maybe `isFirefoxForIOS`.

**Comment by rehandalal** [comment]
I can test on my iDevice.

File: kitsune/sumo/static/sumo/js/aaq.js (hunk 1 above)
Comment (willkg) [inline kitsune/sumo/static/sumo/js/aaq.js]: Maybe `isPersona`?

Comment (mythmon) [inline kitsune/sumo/static/sumo/js/aaq.js]: The first thing I typed was `FirefoxiOS`, and that was even weirder. I couldn't think of anything better. `FirefoxForIOS` sounds like a good one though.

File: kitsune/sumo/static/sumo/js/browserdetect.js (hunk 3 above)
Comment (mythmon) [inline kitsune/sumo/static/sumo/js/browserdetect.js]: Huh, I guess, but only by accident. I wrote this line thinking "I'll add the test for iPod touch, and see how it fails", and then forgot when it didn't fail. I'll add it.

File: kitsune/sumo/static/sumo/js/tests/browserdetecttests.js [hunk 4]
```diff
@@ -89,4 +89,26 @@
     deepEqual(BrowserDetect.detect(ua), ['fxos', 1.3, 'fxos']);
   });
 
+  test('Firefox iOS all platforms', function() {
+    var uas = [
+      'Mozilla/5.0 (iPhone; CPU iPhone OS 8_3 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) FxiOS/1.0 Mobile/12F69 Safari/600.1.4',
+      'Mozilla/5.0 (iPad; CPU iPhone OS 8_3 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) FxiOS/1.0 Mobile/12F69 Safari/600.1.4',
+      'Mozilla/5.0 (iPod touch; CPU iPhone OS 8_4 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) FxiOS/1.0 Mobile/12H143 Safari/600.1.4'
```
Comment (mythmon) [inline kitsune/sumo/static/sumo/js/tests/browserdetecttests.js]: NB: These 3 test cases are the ones that Input uses.

**Comment by rehandalal** [comment]
Want to fix the iPod touch thing?

**Comment by mythmon** [comment]
Oh, oops. Yeah, I'll do that right now.

**Comment by mythmon** [comment]
Feedbacked ^

**Comment by rehandalal** [comment]
cool, r+

**Pull request merged**